app/services/test/slo_response_decoder.rb

```diff
@@ -0,0 +1,37 @@
+module Test
+  class SloResponseDecoder
+    def initialize(params, settings)
+      @params = params
+      @settings = settings
+    end
+
+    def valid_response?
+      response.is_valid? if doc.at_xpath('/samlp:Response', samlp: Saml::XML::Namespaces::PROTOCOL)
+    end
+
+    def valid_signature?
```

Comment:
Perhaps `valid_logout_response?` would be better to make it consistent with `valid_response?`